src/sagemaker/estimator.py

```diff
@@ -442,6 +444,21 @@ def _prepare_for_training(self, job_name=None):
         self._prepare_debugger_for_training()
         self._prepare_profiler_for_training()
 
+    def _validate_framework_rule_compatibility(self, rule):
+        if rule.rule_type != RuleType.DEBUGGER_BUILTIN:
+            # We only validate DEBUGGER BUILTIN RULES for framework compatibility
+            # Since, profiler rules are framework agnostic
+            return True
+        compatible_rules = \
+            set(debugger_rules_utils._get_rule_list(self._framework_name))
+        if rule.name not in compatible_rules:
+            raise RuntimeError(
```

Comment:
Add tests verifying that a RunTimeError is thrown for framework estimator with rules that are incompatible with the framework.